diff --git a/language/proto/generate.go b/language/proto/generate.go
index afa0cbe..48bddc9 100644
--- a/language/proto/generate.go
+++ b/language/proto/generate.go
@@ -235,12 +235,6 @@ func generateProto(pc *ProtoConfig, rel string, pkg *Package, shouldSetVisibilit
 	r.SetPrivateAttr(PackageKey, *pkg)
 	imports := make([]string, 0, len(pkg.Imports))
 	for i := range pkg.Imports {
-		// If FileMode isn't enabled and the proto import is a self import
-		// (an import between the same package), skip it
-		if pc.Mode != FileMode && getPrefix(pc, path.Dir(i)) == getPrefix(pc, rel) {
-			delete(pkg.Imports, i)
-			continue
-		}
 		imports = append(imports, i)
 	}
 	sort.Strings(imports)
